src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -610,53 +968,129 @@ const processAddressUtxos = async (
         diff
       )
 
-      await processor.updateAddressByScriptPubkey(scriptPubkey, {
+      await processor.updateAddressByScriptPubkey(address.scriptPubkey, {
         balance: newBalance,
         used: true
       })
-      await setLookAhead(args)
+      await setLookAhead({ ...args, ...args.path })
     }
   })
 }
 
-interface ProcessRawUtxoArgs extends FormatArgs {
+interface ProcessRawUtxoArgs extends FormatArgs, RawUtxoCacheState {
   utxo: IAccountUTXO
   id: string
   address: Required<IAddress>
 }
 
-const processRawUtxo = async (args: ProcessRawUtxoArgs): Promise<IUTXO> => {
-  const { utxo, id, address, format, walletTools } = args
-
+const processRawUtxo = async (
+  args: ProcessRawUtxoArgs
+): Promise<WsTask<ITransaction> | undefined> => {
+  const {
+    utxo,
+    id,
+    address,
+    format,
+    walletTools,
+    processor,
+    path,
+    taskCache,
+    requiredCount
+  } = args
+  const { rawUtxosCache, processedUtxosCache } = taskCache
   let scriptType: ScriptTypeEnum
   let script: string
   let redeemScript: string | undefined
+
+  let tx: IProcessorTransaction | undefined
   switch (currencyFormatToPurposeType(format)) {
     case BIP43PurposeTypeEnum.Airbitz:
     case BIP43PurposeTypeEnum.Legacy:
-      script = (await fetchTransaction({ ...args, txid: utxo.txid })).hex
       scriptType = ScriptTypeEnum.p2pkh
+
+      // Legacy UTXOs need the previous transaction hex as the script
+      // If we do not currently have it, add it to the queue to fetch it
+      tx = await processor.fetchTransaction(utxo.txid)
+      if (tx == null) {
+        const deferred = new Deferred<ITransaction>()
```

Comment:
Do it everywhere: Rename deferred to transactionTask or something more descriptive.